os-autoinst-obs-auto-submit

```diff
@@ -194,6 +195,45 @@ handle_auto_submit() {
     )
 }
 
+find_latest_prj_version() {
+    local osc_version_findings
+    local versions_array
+    local latest_version="$submit_target_extra"
+    local current_latest_version
+    local -a major_minor_versions
+    local -a latest_major_minor_versions
+
+    current_latest_version="${submit_target_extra##*:}"
+    current_prj="${submit_target_extra%:*}"
+    osc_version_findings=$(osc dists | grep "$current_prj" | awk '{print $5}' | grep -v "$current_latest_version")
+    IFS=$'\n' read -rd '' -a versions_array <<< "$osc_version_findings"
+    if [[ ${#versions_array[@]} -eq 0 ]]; then
+        echo "$submit_target_extra"
+        return 0
+    fi
+
+    # Do not remove the space at the end the curly brackets
+    latest_major_minor_versions=("${current_latest_version//./ }")
+    for version in "${versions_array[@]}"; do
+        major_minor_versions=("${version//./ }")
+
+        # Compare major vers
+        if [[ "${major_minor_versions[0]}" -gt "${latest_major_minor_versions[0]}" ]]; then
+            latest_version="${current_prj}:${version}"
+            latest_major_minor_versions=("${major_minor_versions[@]}")
+        elif [[ "${major_minor_versions[0]}" -eq "${latest_major_minor_versions[0]}" ]]; then
+            # Compare minor vers if majors are equal
+            if [[ "${major_minor_versions[1]}" -gt "${latest_major_minor_versions[1]}" ]]; then
+                latest_version="${current_prj}:${version}"
+                latest_major_minor_versions=("${major_minor_versions[@]}")
+            fi
+        fi
+    done
```

Comment:
Use `sort --version-sort` instead of this.

If you must compare individual versions it is probably better to use rpm (like this `rpm --eval "%{lua:print(rpm.vercmp('1.2-1', '2.0-1'))}"`) but I guess `sort --version-sort` is good enough anyway.

Reply:
Nice. actually super nice. But I will like to try. I didnt know about `sort --version-sort` for example.